### ISSUE TITLE
update of a route table clears entries

### ISSUE THREAD

**Issue by bitglue** [hashicorp/terraform#927]
Given this configuration:

```
variable "cidr_block" {
    description = "Address range for the VPC in CIDR notation"
}

variable "name" {
    description = "Tag the VPC with this name"
}

variable "description" {
    description = "Tag the VPC with this description"
}

variable "region" {
    description = "AWS region to create in"
    default = "us-east-1"
}

output "vpc_id" {
    value = "${aws_vpc.vpc.id}"
}

output "route_table_id" {
    value = "${aws_route_table.public.id}"
}

provider "aws" {
    region = "${var.region}"
}

resource "aws_vpc" "vpc" {
    cidr_block = "${var.cidr_block}"
    tags {
        Name = "${var.name}"
        Description = "${var.description}"
    }
}

resource "aws_internet_gateway" "gateway" {
    vpc_id = "${aws_vpc.vpc.id}"
}

resource "aws_route_table" "public" {
    vpc_id = "${aws_vpc.vpc.id}"

    route {
        cidr_block = "0.0.0.0/0"
        gateway_id = "${aws_internet_gateway.gateway.id}"
    }

    tags {
        Name = "${var.name}"
    }
}
```

The first time this is applied, the VPC and route table are created as expected. If `var.name` is changed, the plan will show each resource being updated in place.

```
~ aws_route_table.public
    tags.Name: "external-dev" => "public-dev"

~ aws_vpc.vpc
    tags.Name: "external-dev" => "public-dev"
```

After this update plan is applied, `terraform.tfstate` will show the route table as having no entries, and `terraform plan` invoked again (with no variables changed) will show them being re-created:

```
~ aws_route_table.public
    route.#:                      "0" => "1"
    route.1446131475.cidr_block:  "" => "0.0.0.0/0"
    route.1446131475.gateway_id:  "" => "igw-ebad178e"
    route.1446131475.instance_id: "" => ""
```

Which it does, but I would not expect the first update to clear the route table's entries.

**Comment by mitchellh** [comment]
Reproduced. Thanks. Taking a look.

**Comment by mitchellh** [comment]
Fixed by #990 once it is merged!
